batcher/in_memory.go

```diff
@@ -3,23 +3,25 @@ package batcher
 import (
 	"sync"
 	"time"
+
+	"github.com/pkg/errors"
 )
 
-type InMemory struct {
-	processor  BatchProcessor
-	buf        []interface{} // protected by mu
-	maxBufSize int
+// A wrapper around a Buffer[Item] that manages thread-safety and flushing.
+type InMemory[Item any, Buf Buffer[Item]] struct {
+	buf Buf // protected by mu
```

Comment:
I don't understand how it benefits us to have InMemory take its buffer implementation as a type argument.  It's already an interface, so why is

```
  buf Buf
```

preferable to

```
  buf Buffer[Item]
```
?

Reply:
I was overly zealous when introducing those type parameters. Fixed in acacff3.